### DISCUSSION TITLE
New user; multi measure rest

### DISCUSSION THREAD

**Discussion by cronebeast48** [w3c-cg/musicxml#459] [Q&A]
I'm trying to use music xml on linux, viewing files with MuseScore 2.
Is there a test suite to validate .musicxml files?

Specific problem: multi measure rest.
  Does this look valid?

  &lt;measure number="27"&gt;
    &lt;attributes&gt;
      &lt;measure-style&gt;
        &lt;multiple-rest&gt;34&lt;/multiple-rest&gt;
      &lt;/measure-style&gt;
    &lt;/attributes&gt;
  &lt;/measure&gt;
  &lt;measure number="61"&gt;

MuseScore 2 doesn't like it.
Tom

**Comment by Jojo-Schmitz** [comment]
Tried MuseScore 3 on it? MuseScore does validate every MusicXML file it imports against the corresponding reference and reports when things don't mix and match, in most cases loads the file if you ignore the error messages. 
Whatever, a full example would be needed, that snippet is not valid MusicXML

**Reply by cronebeast48** [reply]
The example on W3.org/2021/06/musicxml40/musicxml-reference for measure-style/multiple-rest is the
 &lt;attributes&gt;...&lt;/attributes&gt; section above. 
Is it not supposed to be an attribute of a measure?
What makes it invalid?

Tom

**Comment by mdgood** [comment]
You can't just jump from measure 27 to 61 when you have a 34-measure rest. You need `<measure>` elements with a `<rest measure="yes">` for each of bars 28 through 60.

I apologize that the example and documentation does not make this clear. I  have written up issue #460 to improve the documentation to avoid this confusion in the future.

Unfortunately there is no official MusicXML test suite yet. Now that we have full documentation in MusicXML 4.0, maybe a test suite is something that could be worked on for a future version. You can see some pointers to existing test suites in the comments on issue #298.

**Reply by cronebeast48** [reply]
Thank you. 
Wouldn't 34 <measure...><rest measure="yes"></measure>  print 32 measures with 1 measure rests?
How do you get one 34 measure long rest, as shown in the example?
Tom

**Reply by mdgood** [reply]
That's what the `<multiple-rest>` element does. You put it at the start of the first measure of a multimeasure rest, and it instructs the software to draw a multimeasure rest rather than individual 1-measure rests. Semantically there are still 34 measures in that music, so you need 34 measures in the MusicXML file. MusicXML is primarily a semantic format rather than a graphical format, although these days it can capture the graphics with very high accuracy as well.

**Reply by cronebeast48** [reply]
Thank you! That makes sense.
So, to be clear, the example would need to be something like

&lt;measure number="6"&gt;
&lt;attribures&gt;
&lt;measure-style&gt;
&lt;multible-rest&gt;2&lt;/multiple-rest&gt;
&lt;/measure-style&gt;
&lt;/attrubutes&gt;
&lt;rest measure="yes/&gt;
&lt;/measure&gt;
&lt;measure number=7&gt;
&lt;rest measure="yes/&gt;
&lt;/measure&gt;
&lt;measure number=8&gt;
...

I also tried to number the first partial measure as `<measure number="0">` but it seemed to ignore it.

Tom

PS: Before retirement, at cua.edu computer center I used to help write our award winning user guide for the VMS cluster. I would be willing to help with documentation for musicxml as I learn it.

**Reply by cronebeast48** [reply]
That should have been "first partial measure as &lt;measure number="0"&gt;.
(The Mac seems to be dropping all that's in angle brackets; i need the ampersand lt; and gt;)

**Reply by cronebeast48** [reply]
I got it working. It needed 
&lt;measure number="n"&gt;
&lt;note&gt;
&lt;rest/&gt;
&lt;duration&gt;3840&lt;/duration&gt;
&lt;/note&gt;
&lt;/measure&gt;

(and I thought Cobol was wordy!)

**Reply by mdgood** [reply]
You can use back ticks to keep the XML syntax in GitHub text - I edited your comment to restore the `<measure number="0">` that way.

**Comment by cronebeast48** [comment]
I’ve been playing with a musicxml file of the 1st movement of Mahler’s 5th symphony, that was maybe created from an audio file, 
so has lots of artifacts, the worst being the parts for A, B, and C strings and Pizzicato.

I use the page of elements of musicxml for reference, and have a few suggestions:

1. I haven’t found any good explanation of distances.  &lt;scaling&gt; sets the ratio of mm to ’tenths’, ?default 7 / 40, or .175mm/tenth?, 
but many size settings seem to be by hundredths of some norm, 100 being normal size, 50 half, 200 twice. I can’t so far figure out 
what are limitations in MuseScore 2.3, that I’m using to test files, and what is musicxml itself…

2. The documentation examples need comments. Xml could be used but &lt;!— comment —&gt; is pretty wordy.
Using the &lt;staff-distance&gt; example as an example ( :-) ):

&lt;staff-layout <https://www.w3.org/2021/06/musicxml40/musicxml-reference/elements/staff-layout/>&gt;
   &lt;staff-distance <https://www.w3.org/2021/06/musicxml40/musicxml-reference/elements/staff-distance/>&gt;80&lt;/staff-distance <https://www.w3.org/2021/06/musicxml40/musicxml-reference/elements/staff-distance/>&gt;       &lt;!— with scaling 7mm/40tenths, 14mm —&gt;
&lt;/staff-layout <https://www.w3.org/2021/06/musicxml40/musicxml-reference/elements/staff-layout/>&gt;

if I understand things.

Anyway, thanks for your help.

Tom

**Comment by cronebeast48** [comment]
Oops - looks like problem with ‘<‘ etc was on the web site, not email.

I’ve been playing with a musicxml file of the 1st movement of Mahler’s 5th symphony, that was maybe created from an audio file, 
so has lots of artifacts, the worst being the parts for A, B, and C strings and Pizzicato.

I use the page of elements of musicxml for reference, and have a few suggestions:

1. I haven’t found any good explanation of distances.  <scaling> sets the ratio of mm to ’tenths’(a tenth of what?), ?default 7 / 40, or .175mm/tenth?, 
but many size settings seem to be by hundredths of some norm, 100 being normal size, 50 half, 200 twice. I can’t so far figure out 
what are limitations in MuseScore 2.3, that I’m using to test files, and what is musicxml itself…

2. The documentation examples need comments. Xml could be used but <!— comment —> is pretty wordy.
Using the &lt;staff-distance&gt; example as an example  :-) 

<staff-layout>
    <staff-distance>80</staff-distance>     <!— with scaling of 7mm/40tenths, 14mm —>
</staff-layout>

if I understand things…

Anyway, thanks for your help.

Tom

**Comment by cronebeast48** [comment]
I’ve figured out that a tenth is a tenth of the distance between lines on a staff, so scale 7mm to 40 tenths makes a 5 line staff 7mm tall.

It seems odd to define page size in terms of staff height. My page size is set by my printer, and to have to calculate what 8.5 x 11 inches is in terms 
of, say, 5.5/40 millimeters is a pain at best.

Tom
